ui/src/redux/thunks/user.js

```diff
@@ -27,3 +34,19 @@ export const getUserPendingMembers = () => async (dispatch, getState) => {
         dispatch(loadUserPendingMembers(userPendingMembersList, expiry));
     }
 };
+
+export const getUserResourceAccessList = () => async (dispatch, getState) => {
+    let userResourceAccessList = selectUserResourceAccessList(getState());
+    let isUserResourceAccessListEmpty =
+        Array.isArray(userResourceAccessList) &&
+        userResourceAccessList.length < 1;
+    if (isExpired(getState().user.expiry)) {
+        userResourceAccessList = await API().getResourceAccessList({
+            action: 'gcp.assume_role',
```

Comment:
let's not hard code this action here, instead add it as a parameter and let login.js send `gcp.assume_role`, that way other pages can use this if required later on

Reply:
fixed

Reply:
I believe there are few more changes required since currently you are just storing an array. Now if another page makes the same call with a different action, that array may get overridden. So I believe you need to maintain a mapping of "action" to "resourceAccessList". 
Can you confirm @royelbaum

Reply:
yeah i'll need to create a map between the action -> resourceAccessList. Selectors will need changes as well to include action param

Reply:
completely true i talked about it with Jimmy

Reply:
Will address this in a follow up PR.